File: 0217-contains-duplicate/0217-contains-duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_set<int> uset;

        for( auto n : nums){

            if(uset.count(n)){
                return true;
            }
            uset.insert(n);
        }


        return false;
    }
};

File: 0217-contains-duplicate/0217-contains-duplicate.py
class Solution:
    def containsDuplicate(self, nums: list[int]) -> bool:
        # Set to keep track of numbers we have already encountered in the array
        seen = set()
        
        # Iterate through each number in the given list
        for num in nums:
            # Check if the current number is already present in our set
            if num in seen:
                # Return True immediately if a duplicate is found
                return True
            # Add the current number to the set for future checks
            seen.add(num)
            
        # Return False if the loop completes without finding any duplicates
        return False